bundles/org.openhab.binding.mycroft/src/main/java/org/openhab/binding/mycroft/internal/api/MycroftConnection.java

```diff
@@ -0,0 +1,268 @@
+    }
+
+    public void sendMessage(BaseMessage message) throws IOException {
+        sendMessage(gson.toJson(message));
+    }
+
+    public void sendMessage(String message) throws IOException {
+        final Session storedSession = this.session;
+        try {
+            if (storedSession != null) {
+                storedSession.getRemote().sendString(message);
+            } else {
+                throw new IOException("Session is not initialized");
+            }
+        } catch (IOException e) {
+            if (storedSession != null && storedSession.isOpen()) {
+                storedSession.close(-1, "Sending message error");
+            }
+            throw e;
+        }
+    }
+
+    @OnWebSocketConnect
+    public void onConnect(Session session) {
+        connectionState = ConnectionState.CONNECTED;
+        logger.debug("{} successfully connected to {}: {}", socketName, session.getRemoteAddress().getAddress(),
+                session.hashCode());
+        connectionListener.connectionEstablished();
+        this.session = session;
+    }
+
+    @OnWebSocketMessage
+    public void onMessage(Session session, String message) {
+        if (!session.equals(this.session)) {
+            handleWrongSession(session, message);
+            return;
+        }
+        logger.trace("{} received raw data: {}", socketName, message);
+
+        try {
+
+            // listeners on message type :
+            BaseMessage mycroftMessage = gson.fromJson(message, BaseMessage.class);
+            Objects.requireNonNull(mycroftMessage);
+            mycroftMessage.message = message;
+
+            // special listener : to any messages
+            Set<MycroftMessageListener<? extends BaseMessage>> listenersAnyToNotify = listeners.get(MessageType.any);
+            if (listenersAnyToNotify != null && !listenersAnyToNotify.isEmpty()) {
+                for (MycroftMessageListener<? extends BaseMessage> listener : listenersAnyToNotify) {
+                    listener.baseMessageReceived(mycroftMessage);
+                }
+            }
+
+            // listener for specific message type
+            Set<MycroftMessageListener<? extends BaseMessage>> listenersToNotify = listeners.get(mycroftMessage.type);
+            if (mycroftMessage.type != MessageType.any && listenersToNotify != null && !listenersToNotify.isEmpty()) {
+                BaseMessage fullMycroftMessage = gson.fromJson(message, mycroftMessage.type.getMessageTypeClass());
+                mycroftMessage.message = message;
+                Objects.requireNonNull(fullMycroftMessage);
+                listenersToNotify.forEach(listener -> listener.baseMessageReceived(fullMycroftMessage));
+            }
+
+        } catch (RuntimeException e) {
+            // we need to catch all processing exceptions, otherwise they could affect the connection
+            logger.warn("{} encountered an error while processing the message {}: {}", socketName, message,
```

Comment:
What is the impact for the user here? If you log this on the level `warn`, the user will find it in his logs. So is there something he can do about this errors that might occur here? Or are these error some random race condition errors that might happen only sometimes and it doesn't matter if they occur? if it is the later, I would vote for reducing the loglevel to `debug` here and not bother the user if he cannot do anything about the error anyway.

Reply:
I think I copied / pasted this code from another binding with a similar websocket connection (deconz binding) So I'm not sure exactly why this try/catch/warn is here.
Anyway, Done. As you say, it seems logical to pass it to debug.